### PR TITLE
Tests: Popover

### PR DESCRIPTION
**Pull request by amogower**
**What kind of change does this PR introduce?**

Added some tests for the Popover component and sub-components.

I had difficulty determining if there are any useful tests that can be written for the PopoverContainer - open to suggestions!

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: packages/reakit/src/Popover/__tests__/Popover-test.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  expect(getByText("Test")).toHaveAttribute("id", "test");
+  expect(getByText("Test")).toHaveAttribute("aria-label", "test");
+});
+
+test("placement", () => {
+  const { container } = render(<Popover placement="right" />);
+  expect(container.firstChild).toHaveAttribute("data-placement", "right");
+});
+
+// test("flip", () => {
```
Comment (matheus1lva) [inline packages/reakit/src/Popover/__tests__/Popover-test.tsx]: maybe you forgot to uncomment?

File: packages/reakit/src/Popover/__tests__/PopoverArrow-test.tsx [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+  const { getByTestId } = render(
+    <PopoverArrow data-testid="test" aria-label="test" />
+  );
+  expect(getByTestId("test")).toHaveAttribute("aria-label", "test");
+});
+
+test("fillColor", () => {
+  const { container } = render(<PopoverArrow fillColor="red" />);
+  expect(container.firstChild).toMatchInlineSnapshot(`
+.c1 {
```
Comment (matheus1lva) [inline packages/reakit/src/Popover/__tests__/PopoverArrow-test.tsx]: @diegohaz what do you think about putting these long snapshots into their files (i mean, using toMatchSnapshot instead) ?

File: packages/reakit/src/Popover/__tests__/Popover-test.tsx (hunk 1 above)
Comment (amogower) [reply packages/reakit/src/Popover/__tests__/Popover-test.tsx]: Must have!  I'll sort it in the morning

File: packages/reakit/src/Popover/__tests__/PopoverArrow-test.tsx (hunk 2 above)
Comment (amogower) [reply packages/reakit/src/Popover/__tests__/PopoverArrow-test.tsx]: I think that's a good idea in this sort of case, when there are multiple snapshots for style changes

Comment (diegohaz) [reply packages/reakit/src/Popover/__tests__/PopoverArrow-test.tsx]: I agree that long snapshots should be using `toMatchSnapshot`. For the other cases, we should use `toMatchInlineSnapshot` whenever possible.

Comment (matheus1lva) [reply packages/reakit/src/Popover/__tests__/PopoverArrow-test.tsx]: Go for it @amogower !

File: packages/reakit/src/Popover/__tests__/Popover-test.tsx [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+//   const { container } = render(<Popover flip={false} />);
+//   expect(container.firstChild).toHaveAttribute("data-placement", "right");
+// });
+
+// test("shift", () => {
```
Comment (amogower) [inline packages/reakit/src/Popover/__tests__/Popover-test.tsx]: For the "shift" and "flip" props I'm not actually seeing any functionality in Codesandbox - should these be doing something visual @diegohaz @PlayMa256?

Comment (matheus1lva) [reply packages/reakit/src/Popover/__tests__/Popover-test.tsx]: One of them is working, i just don't know, based on the `popper.js` docs, i would say that only shift is reproducible on the playground. It is when you hit the popover with the boundary of your browser, it should change directions. I actually don't know how you could test it...

Comment (amogower) [reply packages/reakit/src/Popover/__tests__/Popover-test.tsx]: Ok I'll dive into Popper.js repo when I get some time and see if I can dig out how they test something like this

Comment (amogower) [reply packages/reakit/src/Popover/__tests__/Popover-test.tsx]: After a bit of research I'm not sure it's particularly trivial to test this so maybe for another time/separate issue?

Comment (diegohaz) [reply packages/reakit/src/Popover/__tests__/Popover-test.tsx]: I guess that's fine. Let's just remove those comments. :)

Comment (amogower) [reply packages/reakit/src/Popover/__tests__/Popover-test.tsx]: @diegohaz Removed them in my last commit so should be good to go!

**Comment by amogower** [comment]
@PlayMa256 @diegohaz Is there anything else I need to do with this?  I'd like to settle on this one and have it merged before I move onto the Tooltip as they are both closely related.

**Comment by PlayMa256** [comment]
Lgtm, only wait for haz to merge it!

**Pull request merged**